src/agent/otubus.cpp

```diff
@@ -56,20 +56,19 @@ namespace BorderRouter {
 namespace ubus {
 static UbusServer *sUbusServerInstance = NULL;
 static int         sUbusEfd            = -1;
+// XXX: do not use static variable to pass parameters
 static void *      sJsonUri            = NULL;
 static int         sBufNum;
 static std::mutex *sNcpThreadMutex;
 
-const static int PANID_LENGTH     = 10;
-const static int XPANID_LENGTH    = 64;
-const static int MASTERKEY_LENGTH = 64;
+const static int PANID_LENGTH     = 10 + 1;
+const static int XPANID_LENGTH    = 64 + 1;
+const static int MASTERKEY_LENGTH = 64 + 1;
 
 UbusServer::UbusServer(Ncp::ControllerOpenThread *aController)
 {
     mController = aController;
     mSecond     = 0;
-    blob_buf_init(&mNetworkdataBuf, 0);
```

Comment:
The init of mNetworkdataBuf is necessary, if not, the first time call "networkdata" function on OpenWRT, it will crash.
The logic of "networkdata" function is: 
    1. send reply through "ubus_send_reply(aContext, aReq, mNetworkdataBuf.head);" (if not init, this will crash)
    2. doing DiagnosticGet function to update the current networkdata. (will cost some time)
    3. Next time ask for networkdata, it will send the latest networkdata.

Reply:
I've moved the initialization to the request handling part, see https://github.com/openthread/ot-br-posix/pull/345/files#diff-44868e5af116c945b79bc922aa59c74fR1348

Reply:
Sorry for ignore that, but in my opinion, the first time to call "networkdata" might still cause crash. The process of "networkdata" is UbusNetworkdataHandler() -> UbusGetInformation() -> "networkdata" branch [https://github.com/openthread/ot-br-posix/blob/master/src/agent/otubus.cpp#L1190](url) -> send mNetworkdataBuf.head (might crash)

Reply:
I've added a guard in the last patch.